server/webapp/WEB-INF/rails.new/spec/controllers/api_v1/admin/environments_controller_spec.rb

```diff
@@ -54,6 +54,26 @@
         expect(controller).to disallow_action(:get, :index).with(401, 'You are not authorized to perform this action.')
       end
     end
+
+    describe :route do
+      describe :with_header do
+        before :each do
+          Rack::MockRequest::DEFAULT_ENV["HTTP_ACCEPT"] = "application/vnd.go.cd.v1+json"
+        end
+        after :each do
+          Rack::MockRequest::DEFAULT_ENV = {}
+        end
```

Comment:
`DEFAULT_ENV` is supposed to contain other keys as well — https://github.com/gocd/gocd/blob/f53cc516f5ef04295fd03dfa9dd8160b0f0745cd/server/webapp/WEB-INF/rails.new/vendor/bundle/jruby/1.9/gems/rack-1.5.5/lib/rack/mock.rb#L43-L50

Might as well include a helper module —

```ruby
module ApiHeaderSetupTeardown
  def self.included(klass)
    klass.class_eval do
      before :each do
        @default_env = Rack::MockRequest::DEFAULT_ENV.dup
        Rack::MockRequest::DEFAULT_ENV.merge!({'HTTP_ACCEPT' => "application/vnd.go.cd.v#{supported_version}+json"})
      end
      after :each do
        Rack::MockRequest::DEFAULT_ENV = @default_env
      end
    end
  end
end

describe FooController do
  include ApiHeaderSetupTeardown
  def supported_version
    1
  end

  it 'should foo' do
    ...
  end
end
```